### ISSUE TITLE
Invalid get index '_category_cmp' (on base: 'Nil')

### ISSUE THREAD

**Issue by Melon-The-Dev** [endlessm/godot-block-coding#94]
### What Happened

i was testing out the plugin, when i downloaded it and added it to my scene no blocks show up on the left side on the block code tab

### How to Reproduce

1. download the plugin and add it to godot 4.2.1
2. enable the plugin
3. add block code node to the scene
4. open block codes tab

### Logs or command-line output

 res://addons/block_code/ui/picker/categories/category_factory.gd:109 - Invalid get index '_category_cmp' (on base: 'Nil').

_No response_

### Anything else?

using godot 4.2.1 and block codes 0.2.0

**Comment by manuq** [comment]
@Melon-The-Dev thank you for the report! I just tried in a empty project using Godot 4.2.2 from Flathub and following the steps I didn't get this error. @dylanmccall you mentioned that you saw this too?

**Comment by wjt** [comment]
@Melon-The-Dev what type of node was the parent of the block code node that you added?

**Comment by dbnicholson** [comment]
Just a guess, but I bet this is [this Godot bug](https://github.com/godotengine/godot/issues/86032) fixed in 4.2.2. That's exactly the pattern we have here. I bet this would be fixed like so:

```
diff --git a/addons/block_code/ui/picker/categories/category_factory.gd b/addons/block_code/ui/picker/categories/category_factory.gd
index fb00181..55e57be 100644
--- a/addons/block_code/ui/picker/categories/category_factory.gd
+++ b/addons/block_code/ui/picker/categories/category_factory.gd
@@ -150,7 +150,7 @@ static func get_categories(blocks: Array[Block], extra_categories: Array[BlockCa
        # convert an array type besides Array.assign().
        var cats: Array[BlockCategory] = []
        cats.assign(cat_map.values())
-       cats.sort_custom(_category_cmp)
+       cats.sort_custom(CategoryFactory._category_cmp)
        return cats
 
 
```

But we could also just raise the minimum version to 4.2.2.

**Comment by dbnicholson** [comment]
Yeah, that simple change fixes the category setup on 4.2.1 and still works on 4.2.2. I'm going to prepare a PR since 4.2.2 has only been out a couple months and there are probably lots of people that haven't upgraded.

**Comment by Melon-The-Dev** [comment]
sorry for the late reply but i found out that yes, the plugin doesnt work on versions before 4.2.2 thanks for the responses tho!
